roachtest: tpcdsvec failed

roachtest.tpcdsvec [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=5442476&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=5442476&tab=artifacts#/tpcdsvec) on release-22.1 @ [3181b7faa2f3b41d6a15ab4b74d2c60bcfe5132d](https://github.com/cockroachdb/cockroach/commits/3181b7faa2f3b41d6a15ab4b74d2c60bcfe5132d):


```
test artifacts and logs in: /artifacts/tpcdsvec/run_1
	tpcdsvec.go:175,tpcdsvec.go:184,test_runner.go:883: ERROR: internal error: concurrent txn use detected. ba: Get [/Table/128/1/13/3204/0,/Min), Get [/Table/128/1/13/14897/0,/Min), Get [/Table/128/1/25/139725/0,/Min), Get [/Table/128/1/42/159564/0,/Min), Get [/Table/128/1/43/3300/0,/Min), Get [/Table/128/1/49/81082/0,/Min), Get [/Table/128/1/80/28841/0,/Min), Get [/Table/128/1/131/106340/0,/Min), Get [/Table/128/1/133/128779/0,/Min), Get [/Table/128/1/147/132488/0,/Min), Get [/Table/128/1/170/56586/0,/Min), Get [/Table/128/1/173/84869/0,/Min), Get [/Table/128/1/236/75875/0,/Min), Get [/Table/128/1/246/142823/0,/Min), Get [/Table/128/1/270/146695/0,/Min), Get [/Table/128/1/278/18256/0,/Min), Get [/Table/128/1/321/105409/0,/Min), Get [/Table/128/1/326/43064/0,/Min), Get [/Table/128/1/361/125338/0,/Min), Get [/Table/128/1/368/8187/0,/Min),... 13 skipped ..., Get [/Table/128/1/575/118456/0,/Min), Get [/Table/128/1/584/12236/0,/Min), Get [/Table/128/1/590/45601/0,/Min), Get [/Table/128/1/598/47521/0,/Min), Get [/Table/128/1/605/66784/0,/Min), [txn: 850b50a4] (SQLSTATE XX000)
		(1) secondary error attachment
		  | ERROR: txn already encountered an error; cannot be used anymore (previous err: concurrent txn use detected. ba: Get [/Table/111/1/28849/0,/Min), Get [/Table/111/1/28862/0,/Min), Get [/Table/111/1/28865/0,/Min), Get [/Table/111/1/28918/0,/Min), Get [/Table/111/1/28954/0,/Min), Get [/Table/111/1/28989/0,/Min), Get [/Table/111/1/29053/0,/Min), Get [/Table/111/1/29074/0,/Min), Get [/Table/111/1/29104/0,/Min), Get [/Table/111/1/29129/0,/Min), Get [/Table/111/1/29141/0,/Min), Get [/Table/111/1/29143/0,/Min), Get [/Table/111/1/29145/0,/Min), Get [/Table/111/1/29150/0,/Min), Get [/Table/111/1/29206/0,/Min), Get [/Table/111/1/29241/0,/Min), Get [/Table/111/1/29256/0,/Min), Get [/Table/111/1/29265/0,/Min), Get [/Table/111/1/29280/0,/Min), Get [/Table/111/1/29292/0,/Min),... 3125 skipped ..., Get [/Table/111/1/75493/0,/Min), Get [/Table/111/1/75506/0,/Min), Get [/Table/111/1/75511/0,/Min), Get [/Table/111/1/75555/0,/Min), Get [/Table/111/1/75595/0,/Min), [txn: 5d94d7bb]) (SQLSTATE XXUUU)
		  | (1)
		  | Wraps: (2) ERROR: txn already encountered an error; cannot be used anymore (previous err: concurrent txn use detected. ba: Get [/Table/111/1/28849/0,/Min), Get [/Table/111/1/28862/0,/Min), Get [/Table/111/1/28865/0,/Min), Get [/Table/111/1/28918/0,/Min), Get [/Table/111/1/28954/0,/Min), Get [/Table/111/1/28989/0,/Min), Get [/Table/111/1/29053/0,/Min), Get [/Table/111/1/29074/0,/Min), Get [/Table/111/1/29104/0,/Min), Get [/Table/111/1/29129/0,/Min), Get [/Table/111/1/29141/0,/Min), Get [/Table/111/1/29143/0,/Min), Get [/Table/111/1/29145/0,/Min), Get [/Table/111/1/29150/0,/Min), Get [/Table/111/1/29206/0,/Min), Get [/Table/111/1/29241/0,/Min), Get [/Table/111/1/29256/0,/Min), Get [/Table/111/1/29265/0,/Min), Get [/Table/111/1/29280/0,/Min), Get [/Table/111/1/29292/0,/Min),... 3125 skipped ..., Get [/Table/111/1/75493/0,/Min), Get [/Table/111/1/75506/0,/Min), Get [/Table/111/1/75511/0,/Min), Get [/Table/111/1/75555/0,/Min), Get [/Table/111/1/75595/0,/Min), [txn: 5d94d7bb]) (SQLSTATE XXUUU)
		  | Error types: (1) *fmt.wrapError (2) *pgconn.PgError
		Wraps: (2) secondary error attachment
		  | ERROR: internal error: concurrent txn use detected. ba: Scan [/Table/124/1/2451047/1,/Table/124/1/2451047/2), Scan [/Table/124/1/2451047/10,/Table/124/1/2451047/11), Scan [/Table/124/1/2451047/182,/Table/124/1/2451047/183), Scan [/Table/124/1/2451047/226,/Table/124/1/2451047/227), Scan [/Table/124/1/2451047/235,/Table/124/1/2451047/236), Scan [/Table/124/1/2451047/316,/Table/124/1/2451047/317), Scan [/Table/124/1/2451047/440,/Table/124/1/2451047/441), Scan [/Table/124/1/2451047/458,/Table/124/1/2451047/459), Scan [/Table/124/1/2451047/511,/Table/124/1/2451047/511/PrefixEnd), Scan [/Table/124/1/2451047/529,/Table/124/1/2451047/530), Scan [/Table/124/1/2451047/542,/Table/124/1/2451047/543), Scan [/Table/124/1/2451047/586,/Table/124/1/2451047/587), Scan [/Table/124/1/2451047/592,/Table/124/1/2451047/593), Scan [/Table/124/1/2451047/607,/Table/124/1/2451047/608), Scan [/Table/124/1/2451047/608,/Table/124/1/2451047/609), Scan [/Table/124/1/2451047/637,/Table/124/1/2451047/638), Scan [/Table/124/1/2451047/650,/Table/124/1/2451047/651), Scan [/Table/124/1/2451047/662,/Table/124/1/2451047/663), Scan [/Table/124/1/2451047/676,/Table/124/1/2451047/677), Scan [/Table/124/1/2451047/692,/Table/124/1/2451047/693),... 1921 skipped ..., Scan [/Table/124/1/2451049/17797,/Table/124/1/2451049/17798), Scan [/Table/124/1/2451049/17827,/Table/124/1/2451049/17828), Scan [/Table/124/1/2451049/17858,/Table/124/1/2451049/17859), Scan [/Table/124/1/2451049/17954,/Table/124/1/2451049/17955), Scan [/Table/124/1/2451049/17971,/Table/124/1/2451049/17972), [txn: 6cd46fc0] (SQLSTATE XX000)
		  | (1)
		  | Wraps: (2) ERROR: internal error: concurrent txn use detected. ba: Scan [/Table/124/1/2451047/1,/Table/124/1/2451047/2), Scan [/Table/124/1/2451047/10,/Table/124/1/2451047/11), Scan [/Table/124/1/2451047/182,/Table/124/1/2451047/183), Scan [/Table/124/1/2451047/226,/Table/124/1/2451047/227), Scan [/Table/124/1/2451047/235,/Table/124/1/2451047/236), Scan [/Table/124/1/2451047/316,/Table/124/1/2451047/317), Scan [/Table/124/1/2451047/440,/Table/124/1/2451047/441), Scan [/Table/124/1/2451047/458,/Table/124/1/2451047/459), Scan [/Table/124/1/2451047/511,/Table/124/1/2451047/511/PrefixEnd), Scan [/Table/124/1/2451047/529,/Table/124/1/2451047/530), Scan [/Table/124/1/2451047/542,/Table/124/1/2451047/543), Scan [/Table/124/1/2451047/586,/Table/124/1/2451047/587), Scan [/Table/124/1/2451047/592,/Table/124/1/2451047/593), Scan [/Table/124/1/2451047/607,/Table/124/1/2451047/608), Scan [/Table/124/1/2451047/608,/Table/124/1/2451047/609), Scan [/Table/124/1/2451047/637,/Table/124/1/2451047/638), Scan [/Table/124/1/2451047/650,/Table/124/1/2451047/651), Scan [/Table/124/1/2451047/662,/Table/124/1/2451047/663), Scan [/Table/124/1/2451047/676,/Table/124/1/2451047/677), Scan [/Table/124/1/2451047/692,/Table/124/1/2451047/693),... 1921 skipped ..., Scan [/Table/124/1/2451049/17797,/Table/124/1/2451049/17798), Scan [/Table/124/1/2451049/17827,/Table/124/1/2451049/17828), Scan [/Table/124/1/2451049/17858,/Table/124/1/2451049/17859), Scan [/Table/124/1/2451049/17954,/Table/124/1/2451049/17955), Scan [/Table/124/1/2451049/17971,/Table/124/1/2451049/17972), [txn: 6cd46fc0] (SQLSTATE XX000)
		  | Error types: (1) *fmt.wrapError (2) *pgconn.PgError
		Wraps: (3) secondary error attachment
		  | ERROR: internal error: concurrent txn use detected. ba: Scan [/Table/129/1/11976/120300/0,/Table/129/1/11977), Scan [/Table/129/1/12014,/Table/129/1/12015), Scan [/Table/129/1/12016,/Table/129/1/12017), Scan [/Table/129/1/12038,/Table/129/1/12039), Scan [/Table/129/1/12094,/Table/129/1/12095), Scan [/Table/129/1/12116,/Table/129/1/12117), Scan [/Table/129/1/12132,/Table/129/1/12133), Scan [/Table/129/1/12146,/Table/129/1/12147), Scan [/Table/129/1/12150,/Table/129/1/12151), Scan [/Table/129/1/12172,/Table/129/1/12173), Scan [/Table/129/1/12178,/Table/129/1/12179), Scan [/Table/129/1/12186,/Table/129/1/12187), Scan [/Table/129/1/12216,/Table/129/1/12217), Scan [/Table/129/1/12258,/Table/129/1/12259), Scan [/Table/129/1/12308,/Table/129/1/12309), Scan [/Table/129/1/12338,/Table/129/1/12339), Scan [/Table/129/1/12354,/Table/129/1/12355), Scan [/Table/129/1/12358,/Table/129/1/12359), Scan [/Table/129/1/12400,/Table/129/1/12401), Scan [/Table/129/1/12470,/Table/129/1/12471),... 297 skipped ..., Scan [/Table/129/1/17924,/Table/129/1/17925), Scan [/Table/129/1/17950,/Table/129/1/17951), Scan [/Table/129/1/17958,/Table/129/1/17959), Scan [/Table/129/1/17962,/Table/129/1/17963), Scan [/Table/129/1/17980,/Table/129/1/17981), [txn: 89bb0ca0] (SQLSTATE XX000)
		  | (1)
		  | Wraps: (2) ERROR: internal error: concurrent txn use detected. ba: Scan [/Table/129/1/11976/120300/0,/Table/129/1/11977), Scan [/Table/129/1/12014,/Table/129/1/12015), Scan [/Table/129/1/12016,/Table/129/1/12017), Scan [/Table/129/1/12038,/Table/129/1/12039), Scan [/Table/129/1/12094,/Table/129/1/12095), Scan [/Table/129/1/12116,/Table/129/1/12117), Scan [/Table/129/1/12132,/Table/129/1/12133), Scan [/Table/129/1/12146,/Table/129/1/12147), Scan [/Table/129/1/12150,/Table/129/1/12151), Scan [/Table/129/1/12172,/Table/129/1/12173), Scan [/Table/129/1/12178,/Table/129/1/12179), Scan [/Table/129/1/12186,/Table/129/1/12187), Scan [/Table/129/1/12216,/Table/129/1/12217), Scan [/Table/129/1/12258,/Table/129/1/12259), Scan [/Table/129/1/12308,/Table/129/1/12309), Scan [/Table/129/1/12338,/Table/129/1/12339), Scan [/Table/129/1/12354,/Table/129/1/12355), Scan [/Table/129/1/12358,/Table/129/1/12359), Scan [/Table/129/1/12400,/Table/129/1/12401), Scan [/Table/129/1/12470,/Table/129/1/12471),... 297 skipped ..., Scan [/Table/129/1/17924,/Table/129/1/17925), Scan [/Table/129/1/17950,/Table/129/1/17951), Scan [/Table/129/1/17958,/Table/129/1/17959), Scan [/Table/129/1/17962,/Table/129/1/17963), Scan [/Table/129/1/17980,/Table/129/1/17981), [txn: 89bb0ca0] (SQLSTATE XX000)
		  | Error types: (1) *fmt.wrapError (2) *pgconn.PgError
		Wraps: (4) secondary error attachment
		  | ERROR: internal error: concurrent txn use detected. ba: Scan [/Table/129/1/3,/Table/129/1/4), Scan [/Table/129/1/12,/Table/129/1/13), Scan [/Table/129/1/32,/Table/129/1/33), Scan [/Table/129/1/40,/Table/129/1/41), Scan [/Table/129/1/109,/Table/129/1/109/PrefixEnd), Scan [/Table/129/1/114,/Table/129/1/115), Scan [/Table/129/1/115,/Table/129/1/116), Scan [/Table/129/1/127,/Table/129/1/128), Scan [/Table/129/1/128,/Table/129/1/129), Scan [/Table/129/1/144,/Table/129/1/145), Scan [/Table/129/1/155,/Table/129/1/156), Scan [/Table/129/1/158,/Table/129/1/159), Scan [/Table/129/1/229,/Table/129/1/230), Scan [/Table/129/1/238,/Table/129/1/239), Scan [/Table/129/1/252,/Table/129/1/253), Scan [/Table/129/1/266,/Table/129/1/267), Scan [/Table/129/1/272,/Table/129/1/273), Scan [/Table/129/1/274,/Table/129/1/275), Scan [/Table/129/1/282,/Table/129/1/283), Scan [/Table/129/1/283,/Table/129/1/284),... 1782 skipped ..., Scan [/Table/129/1/17943,/Table/129/1/17944), Scan [/Table/129/1/17967,/Table/129/1/17968), Scan [/Table/129/1/17972,/Table/129/1/17973), Scan [/Table/129/1/17974,/Table/129/1/17975), Scan [/Table/129/1/17987,/Table/129/1/17988), [txn: ce8d8619] (SQLSTATE XX000)
		  | (1)
		  | Wraps: (2) ERROR: internal error: concurrent txn use detected. ba: Scan [/Table/129/1/3,/Table/129/1/4), Scan [/Table/129/1/12,/Table/129/1/13), Scan [/Table/129/1/32,/Table/129/1/33), Scan [/Table/129/1/40,/Table/129/1/41), Scan [/Table/129/1/109,/Table/129/1/109/PrefixEnd), Scan [/Table/129/1/114,/Table/129/1/115), Scan [/Table/129/1/115,/Table/129/1/116), Scan [/Table/129/1/127,/Table/129/1/128), Scan [/Table/129/1/128,/Table/129/1/129), Scan [/Table/129/1/144,/Table/129/1/145), Scan [/Table/129/1/155,/Table/129/1/156), Scan [/Table/129/1/158,/Table/129/1/159), Scan [/Table/129/1/229,/Table/129/1/230), Scan [/Table/129/1/238,/Table/129/1/239), Scan [/Table/129/1/252,/Table/129/1/253), Scan [/Table/129/1/266,/Table/129/1/267), Scan [/Table/129/1/272,/Table/129/1/273), Scan [/Table/129/1/274,/Table/129/1/275), Scan [/Table/129/1/282,/Table/129/1/283), Scan [/Table/129/1/283,/Table/129/1/284),... 1782 skipped ..., Scan [/Table/129/1/17943,/Table/129/1/17944), Scan [/Table/129/1/17967,/Table/129/1/17968), Scan [/Table/129/1/17972,/Table/129/1/17973), Scan [/Table/129/1/17974,/Table/129/1/17975), Scan [/Table/129/1/17987,/Table/129/1/17988), [txn: ce8d8619] (SQLSTATE XX000)
		  | Error types: (1) *fmt.wrapError (2) *pgconn.PgError
		Wraps: (5) secondary error attachment
		  | ERROR: internal error: concurrent txn use detected. ba: Get [/Table/108/1/2450816/0,/Min), Get [/Table/108/1/2450818/0,/Min), Get [/Table/108/1/2450819/0,/Min), Get [/Table/108/1/2450820/0,/Min), Get [/Table/108/1/2450821/0,/Min), Get [/Table/108/1/2450822/0,/Min), Get [/Table/108/1/2450825/0,/Min), Get [/Table/108/1/2450826/0,/Min), Get [/Table/108/1/2450827/0,/Min), Get [/Table/108/1/2450828/0,/Min), Get [/Table/108/1/2450829/0,/Min), Get [/Table/108/1/2450830/0,/Min), Get [/Table/108/1/2450831/0,/Min), Get [/Table/108/1/2450832/0,/Min), Get [/Table/108/1/2450833/0,/Min), Get [/Table/108/1/2450834/0,/Min), Get [/Table/108/1/2450836/0,/Min), Get [/Table/108/1/2450837/0,/Min), Get [/Table/108/1/2450838/0,/Min), Get [/Table/108/1/2450840/0,/Min),... 1178 skipped ..., Get [/Table/108/1/2452638/0,/Min), Get [/Table/108/1/2452639/0,/Min), Get [/Table/108/1/2452640/0,/Min), Get [/Table/108/1/2452641/0,/Min), Get [/Table/108/1/2452642/0,/Min), [txn: a11b9b67] (SQLSTATE XX000)
		  | (1)
		  | Wraps: (2) ERROR: internal error: concurrent txn use detected. ba: Get [/Table/108/1/2450816/0,/Min), Get [/Table/108/1/2450818/0,/Min), Get [/Table/108/1/2450819/0,/Min), Get [/Table/108/1/2450820/0,/Min), Get [/Table/108/1/2450821/0,/Min), Get [/Table/108/1/2450822/0,/Min), Get [/Table/108/1/2450825/0,/Min), Get [/Table/108/1/2450826/0,/Min), Get [/Table/108/1/2450827/0,/Min), Get [/Table/108/1/2450828/0,/Min), Get [/Table/108/1/2450829/0,/Min), Get [/Table/108/1/2450830/0,/Min), Get [/Table/108/1/2450831/0,/Min), Get [/Table/108/1/2450832/0,/Min), Get [/Table/108/1/2450833/0,/Min), Get [/Table/108/1/2450834/0,/Min), Get [/Table/108/1/2450836/0,/Min), Get [/Table/108/1/2450837/0,/Min), Get [/Table/108/1/2450838/0,/Min), Get [/Table/108/1/2450840/0,/Min),... 1178 skipped ..., Get [/Table/108/1/2452638/0,/Min), Get [/Table/108/1/2452639/0,/Min), Get [/Table/108/1/2452640/0,/Min), Get [/Table/108/1/2452641/0,/Min), Get [/Table/108/1/2452642/0,/Min), [txn: a11b9b67] (SQLSTATE XX000)
		  | Error types: (1) *fmt.wrapError (2) *pgconn.PgError
		Wraps: (6) secondary error attachment
		  | ERROR: internal error: concurrent txn use detected. ba: Get [/Table/108/1/2450815/0,/Min), Get [/Table/108/1/2450816/0,/Min), Get [/Table/108/1/2450817/0,/Min), Get [/Table/108/1/2450818/0,/Min), Get [/Table/108/1/2450819/0,/Min), Get [/Table/108/1/2450820/0,/Min), Get [/Table/108/1/2450822/0,/Min), Get [/Table/108/1/2450824/0,/Min), Get [/Table/108/1/2450825/0,/Min), Get [/Table/108/1/2450828/0,/Min), Get [/Table/108/1/2450829/0,/Min), Get [/Table/108/1/2450830/0,/Min), Get [/Table/108/1/2450832/0,/Min), Get [/Table/108/1/2450834/0,/Min), Get [/Table/108/1/2450835/0,/Min), Get [/Table/108/1/2450837/0,/Min), Get [/Table/108/1/2450840/0,/Min), Get [/Table/108/1/2450841/0,/Min), Get [/Table/108/1/2450842/0,/Min), Get [/Table/108/1/2450844/0,/Min),... 1371 skipped ..., Get [/Table/108/1/2452639/0,/Min), Get [/Table/108/1/2452640/0,/Min), Get [/Table/108/1/2452641/0,/Min), Get [/Table/108/1/2452644/0,/Min), Get [/Table/108/1/2452646/0,/Min), [txn: 6a23b811] (SQLSTATE XX000)
		  | (1)
		  | Wraps: (2) ERROR: internal error: concurrent txn use detected. ba: Get [/Table/108/1/2450815/0,/Min), Get [/Table/108/1/2450816/0,/Min), Get [/Table/108/1/2450817/0,/Min), Get [/Table/108/1/2450818/0,/Min), Get [/Table/108/1/2450819/0,/Min), Get [/Table/108/1/2450820/0,/Min), Get [/Table/108/1/2450822/0,/Min), Get [/Table/108/1/2450824/0,/Min), Get [/Table/108/1/2450825/0,/Min), Get [/Table/108/1/2450828/0,/Min), Get [/Table/108/1/2450829/0,/Min), Get [/Table/108/1/2450830/0,/Min), Get [/Table/108/1/2450832/0,/Min), Get [/Table/108/1/2450834/0,/Min), Get [/Table/108/1/2450835/0,/Min), Get [/Table/108/1/2450837/0,/Min), Get [/Table/108/1/2450840/0,/Min), Get [/Table/108/1/2450841/0,/Min), Get [/Table/108/1/2450842/0,/Min), Get [/Table/108/1/2450844/0,/Min),... 1371 skipped ..., Get [/Table/108/1/2452639/0,/Min), Get [/Table/108/1/2452640/0,/Min), Get [/Table/108/1/2452641/0,/Min), Get [/Table/108/1/2452644/0,/Min), Get [/Table/108/1/2452646/0,/Min), [txn: 6a23b811] (SQLSTATE XX000)
		  | Error types: (1) *fmt.wrapError (2) *pgconn.PgError
		Wraps: (7) secondary error attachment
		  | ERROR: txn already encountered an error; cannot be used anymore (previous err: concurrent txn use detected. ba: Get [/Table/111/1/28802/0,/Min), Get [/Table/111/1/28820/0,/Min), Get [/Table/111/1/28881/0,/Min), Get [/Table/111/1/28883/0,/Min), Get [/Table/111/1/28897/0,/Min), Get [/Table/111/1/28909/0,/Min), Get [/Table/111/1/28943/0,/Min), Get [/Table/111/1/28950/0,/Min), Get [/Table/111/1/28963/0,/Min), Get [/Table/111/1/28975/0,/Min), Get [/Table/111/1/28989/0,/Min), Get [/Table/111/1/29004/0,/Min), Get [/Table/111/1/29039/0,/Min), Get [/Table/111/1/29053/0,/Min), Get [/Table/111/1/29054/0,/Min), Get [/Table/111/1/29140/0,/Min), Get [/Table/111/1/29145/0,/Min), Get [/Table/111/1/29165/0,/Min), Get [/Table/111/1/29182/0,/Min), Get [/Table/111/1/29199/0,/Min),... 4402 skipped ..., Get [/Table/111/1/75507/0,/Min), Get [/Table/111/1/75567/0,/Min), Get [/Table/111/1/75568/0,/Min), Get [/Table/111/1/75595/0,/Min), Get [/Table/111/1/75598/0,/Min), [txn: bced45d4], [can-forward-ts]) (SQLSTATE XXUUU)
		  | (1)
		  | Wraps: (2) ERROR: txn already encountered an error; cannot be used anymore (previous err: concurrent txn use detected. ba: Get [/Table/111/1/28802/0,/Min), Get [/Table/111/1/28820/0,/Min), Get [/Table/111/1/28881/0,/Min), Get [/Table/111/1/28883/0,/Min), Get [/Table/111/1/28897/0,/Min), Get [/Table/111/1/28909/0,/Min), Get [/Table/111/1/28943/0,/Min), Get [/Table/111/1/28950/0,/Min), Get [/Table/111/1/28963/0,/Min), Get [/Table/111/1/28975/0,/Min), Get [/Table/111/1/28989/0,/Min), Get [/Table/111/1/29004/0,/Min), Get [/Table/111/1/29039/0,/Min), Get [/Table/111/1/29053/0,/Min), Get [/Table/111/1/29054/0,/Min), Get [/Table/111/1/29140/0,/Min), Get [/Table/111/1/29145/0,/Min), Get [/Table/111/1/29165/0,/Min), Get [/Table/111/1/29182/0,/Min), Get [/Table/111/1/29199/0,/Min),... 4402 skipped ..., Get [/Table/111/1/75507/0,/Min), Get [/Table/111/1/75567/0,/Min), Get [/Table/111/1/75568/0,/Min), Get [/Table/111/1/75595/0,/Min), Get [/Table/111/1/75598/0,/Min), [txn: bced45d4], [can-forward-ts]) (SQLSTATE XXUUU)
		  | Error types: (1) *fmt.wrapError (2) *pgconn.PgError
		Wraps: (8) secondary error attachment
		  | ERROR: txn already encountered an error; cannot be used anymore (previous err: concurrent txn use detected. ba: Get [/Table/111/1/18270/0,/Min), Get [/Table/111/1/23680/0,/Min), Get [/Table/111/1/24958/0,/Min), Get [/Table/111/1/45135/0,/Min), Get [/Table/111/1/53438/0,/Min), Get [/Table/111/1/74825/0,/Min), Get [/Table/111/1/75289/0,/Min), Get [/Table/111/1/78885/0,/Min), [txn: 407325da]) (SQLSTATE XXUUU)
		  | (1)
		  | Wraps: (2) ERROR: txn already encountered an error; cannot be used anymore (previous err: concurrent txn use detected. ba: Get [/Table/111/1/18270/0,/Min), Get [/Table/111/1/23680/0,/Min), Get [/Table/111/1/24958/0,/Min), Get [/Table/111/1/45135/0,/Min), Get [/Table/111/1/53438/0,/Min), Get [/Table/111/1/74825/0,/Min), Get [/Table/111/1/75289/0,/Min), Get [/Table/111/1/78885/0,/Min), [txn: 407325da]) (SQLSTATE XXUUU)
		  | Error types: (1) *fmt.wrapError (2) *pgconn.PgError
		Wraps: (9) secondary error attachment
		  | ERROR: internal error: concurrent txn use detected. ba: Get [/Table/106/1/11/0,/Min), Get [/Table/106/1/20/0,/Min), Get [/Table/106/1/38/0,/Min), Get [/Table/106/1/42/0,/Min), Get [/Table/106/1/80/0,/Min), Get [/Table/106/1/140/0,/Min), Get [/Table/106/1/142/0,/Min), Get [/Table/106/1/286/0,/Min), Get [/Table/106/1/304/0,/Min), Get [/Table/106/1/315/0,/Min), Get [/Table/106/1/330/0,/Min), Get [/Table/106/1/362/0,/Min), Get [/Table/106/1/396/0,/Min), Get [/Table/106/1/442/0,/Min), Get [/Table/106/1/451/0,/Min), Get [/Table/106/1/488/0,/Min), Get [/Table/106/1/494/0,/Min), Get [/Table/106/1/498/0,/Min), Get [/Table/106/1/503/0,/Min), Get [/Table/106/1/538/0,/Min),... 1579 skipped ..., Get [/Table/106/1/49854/0,/Min), Get [/Table/106/1/49874/0,/Min), Get [/Table/106/1/49885/0,/Min), Get [/Table/106/1/49889/0,/Min), Get [/Table/106/1/49992/0,/Min), [txn: 962731c0], [can-forward-ts] (SQLSTATE XX000)
		  | (1)
		  | Wraps: (2) ERROR: internal error: concurrent txn use detected. ba: Get [/Table/106/1/11/0,/Min), Get [/Table/106/1/20/0,/Min), Get [/Table/106/1/38/0,/Min), Get [/Table/106/1/42/0,/Min), Get [/Table/106/1/80/0,/Min), Get [/Table/106/1/140/0,/Min), Get [/Table/106/1/142/0,/Min), Get [/Table/106/1/286/0,/Min), Get [/Table/106/1/304/0,/Min), Get [/Table/106/1/315/0,/Min), Get [/Table/106/1/330/0,/Min), Get [/Table/106/1/362/0,/Min), Get [/Table/106/1/396/0,/Min), Get [/Table/106/1/442/0,/Min), Get [/Table/106/1/451/0,/Min), Get [/Table/106/1/488/0,/Min), Get [/Table/106/1/494/0,/Min), Get [/Table/106/1/498/0,/Min), Get [/Table/106/1/503/0,/Min), Get [/Table/106/1/538/0,/Min),... 1579 skipped ..., Get [/Table/106/1/49854/0,/Min), Get [/Table/106/1/49874/0,/Min), Get [/Table/106/1/49885/0,/Min), Get [/Table/106/1/49889/0,/Min), Get [/Table/106/1/49992/0,/Min), [txn: 962731c0], [can-forward-ts] (SQLSTATE XX000)
		  | Error types: (1) *fmt.wrapError (2) *pgconn.PgError
		Wraps: (10)
		Wraps: (11) ERROR: internal error: concurrent txn use detected. ba: Get [/Table/128/1/13/3204/0,/Min), Get [/Table/128/1/13/14897/0,/Min), Get [/Table/128/1/25/139725/0,/Min), Get [/Table/128/1/42/159564/0,/Min), Get [/Table/128/1/43/3300/0,/Min), Get [/Table/128/1/49/81082/0,/Min), Get [/Table/128/1/80/28841/0,/Min), Get [/Table/128/1/131/106340/0,/Min), Get [/Table/128/1/133/128779/0,/Min), Get [/Table/128/1/147/132488/0,/Min), Get [/Table/128/1/170/56586/0,/Min), Get [/Table/128/1/173/84869/0,/Min), Get [/Table/128/1/236/75875/0,/Min), Get [/Table/128/1/246/142823/0,/Min), Get [/Table/128/1/270/146695/0,/Min), Get [/Table/128/1/278/18256/0,/Min), Get [/Table/128/1/321/105409/0,/Min), Get [/Table/128/1/326/43064/0,/Min), Get [/Table/128/1/361/125338/0,/Min), Get [/Table/128/1/368/8187/0,/Min),... 13 skipped ..., Get [/Table/128/1/575/118456/0,/Min), Get [/Table/128/1/584/12236/0,/Min), Get [/Table/128/1/590/45601/0,/Min), Get [/Table/128/1/598/47521/0,/Min), Get [/Table/128/1/605/66784/0,/Min), [txn: 850b50a4] (SQLSTATE XX000)
		Error types: (1) *secondary.withSecondaryError (2) *secondary.withSecondaryError (3) *secondary.withSecondaryError (4) *secondary.withSecondaryError (5) *secondary.withSecondaryError (6) *secondary.withSecondaryError (7) *secondary.withSecondaryError (8) *secondary.withSecondaryError (9) *secondary.withSecondaryError (10) *fmt.wrapError (11) *pgconn.PgError
```
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)

</p>
</details>
/cc @cockroachdb/sql-queries
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*tpcdsvec.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-16652

--- a/pkg/sql/distsql/server.go
+++ b/pkg/sql/distsql/server.go
@@ -298,6 +298,10 @@ func (ds *ServerImpl) setupFlow(
 	var evalCtx *eval.Context
 	var leafTxn *kv.Txn
 	if localState.EvalContext != nil {
+		// If we're running on the gateway, then we'll reuse already existing
+		// eval context. This is the case even if the query is distributed -
+		// this allows us to avoid an unnecessary deserialization of the eval
+		// context proto.
 		evalCtx = localState.EvalContext
 		// We're about to mutate the evalCtx and we want to restore its original
 		// state once the flow cleans up. Note that we could have made a copy of
@@ -310,7 +314,7 @@ func (ds *ServerImpl) setupFlow(
 			evalCtx.Txn = origTxn
 		}
 		evalCtx.Mon = monitor
-		if localState.HasConcurrency {
+		if localState.MustUseLeafTxn() {
 			var err error
 			leafTxn, err = makeLeaf()
 			if err != nil {

